Fix path in generated `env.d.ts` on Windows

### Describe the bug

Wrong import path results in `any` type. It should be:

```diff
- import('..\src\env.ts')
+ import('../src/env.ts')
```

> Cannot find module '..srcenv.ts' or its corresponding type declarations.

```ts
// .svelte-kit\env.d.ts

// This file is generated from ..\src\env.ts.
// See https://svelte.dev/docs/kit/environment-variables for more information

declare module '$app/env/private' {
  export const DATABASE_URL: import('@sveltejs/kit/internal/types').StandardSchemaV1.InferOutput<
    // Cannot find module '..srcenv.ts' or its corresponding type declarations.
    typeof import('..\src\env.ts').variables.DATABASE_URL.schema
  >;
}

declare module '$app/env/public' {
  export const SENTRY_DSN: import('@sveltejs/kit/internal/types').StandardSchemaV1.InferOutput<
    // Cannot find module '..srcenv.ts' or its corresponding type declarations.
    typeof import('..\src\env.ts').variables.SENTRY_DSN.schema
  >;
}
```

### Reproduction

.

### Logs

```Shell

```

### System Info

```Shell
System:
  OS: Windows 11 10.0.26200
  CPU: (8) x64 Intel(R) Core(TM) Ultra 7 258V
  Memory: 2.91 GB / 31.48 GB
Binaries:
  Node: 24.16.0 - C:\Users\hyunb\AppData\Local\fnm_multishells\2836_1780667384993\node.EXE
  Yarn: 1.22.22 - C:\Users\hyunb\AppData\Local\fnm_multishells\2836_1780667384993\yarn.CMD
  npm: 11.13.0 - C:\Users\hyunb\AppData\Local\fnm_multishells\2836_1780667384993\npm.CMD
  pnpm: 11.5.2 - C:\Users\hyunb\AppData\Local\fnm_multishells\2836_1780667384993\pnpm.CMD
  Deno: 2.4.0 - C:\Users\hyunb\.deno\bin\deno.EXE
Browsers:
  Chrome: 148.0.7778.217
  Edge: Chromium (140.0.3485.54)
  Firefox: 151.0.3 - C:\Program Files\Mozilla Firefox\firefox.exe
npmPackages:
  @sveltejs/adapter-node: ^5.5.4 => 5.5.4
  @sveltejs/kit: ^2.63.0 => 2.63.0
  @sveltejs/vite-plugin-svelte: ^7.1.2 => 7.1.2
  svelte: ^5.56.1 => 5.56.1
  vite: ^8.0.14 => 8.0.14
```

### Severity

blocking an upgrade

### Additional Information

_No response_

--- a/.changeset/fix-env-dts-windows-path.md
+++ b/.changeset/fix-env-dts-windows-path.md
@@ -1,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+fix: use forward slashes in the generated `env.d.ts` import path on Windows

--- a/packages/kit/src/core/sync/write_env.js
+++ b/packages/kit/src/core/sync/write_env.js
@@ -1,6 +1,7 @@
 /** @import { EnvVarConfig } from '@sveltejs/kit' */
 import path from 'node:path';
 import { create_explicit_env_types } from '../env.js';
+import { posixify } from '../../utils/filesystem.js';
 import { write_if_changed } from './utils.js';
 
 const DOCS = '// See https://svelte.dev/docs/kit/environment-variables for more information';
@@ -18,7 +19,7 @@ export function write_env(kit, entry, env_config) {
 	const out = path.join(kit.outDir, 'env.d.ts');
 
 	if (entry && env_config) {
-		const relative = path.relative(kit.outDir, entry);
+		const relative = posixify(path.relative(kit.outDir, entry));
 		content.push(
 			`// This file is generated from ${relative}.\n${DOCS}`,
 			create_explicit_env_types(env_config, relative, 'private'),

